mm2src/mm2_rpc/src/data/legacy/orders.rs

```diff
@@ -95,106 +197,309 @@ impl Default for MatchBy {
+    pub pairs: Vec<(String, String)>,
+}
 
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct RpcOrderbookEntry {
-    pub coin: String,
-    pub address: String,
-    pub price: BigDecimal,
-    pub price_rat: BigRational,
-    pub price_fraction: Fraction,
-    #[serde(rename = "maxvolume")]
-    pub max_volume: BigDecimal,
-    pub max_volume_rat: BigRational,
-    pub max_volume_fraction: Fraction,
-    pub min_volume: BigDecimal,
-    pub min_volume_rat: BigRational,
-    pub min_volume_fraction: Fraction,
-    pub pubkey: String,
-    pub age: u64,
-    pub uuid: Uuid,
-    pub is_mine: bool,
-    #[serde(flatten)]
-    pub base_max_volume: DetailedBaseMaxVolume,
-    #[serde(flatten)]
-    pub base_min_volume: DetailedBaseMinVolume,
-    #[serde(flatten)]
-    pub rel_max_volume: DetailedRelMaxVolume,
-    #[serde(flatten)]
-    pub rel_min_volume: DetailedRelMinVolume,
-    #[serde(flatten)]
-    pub conf_settings: Option<OrderConfirmationsSettings>,
+#[derive(Debug, Deserialize, Serialize)]
+pub struct PairWithDepth {
+    pub pair: (String, String),
+    pub depth: PairDepth,
 }
 
-construct_detailed!(DetailedBaseMaxVolume, base_max_volume);
-construct_detailed!(DetailedBaseMinVolume, base_min_volume);
-construct_detailed!(DetailedRelMaxVolume, rel_max_volume);
-construct_detailed!(DetailedRelMinVolume, rel_min_volume);
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct PairDepth {
+    pub asks: usize,
+    pub bids: usize,
+}
 
 #[derive(Debug, Serialize, Deserialize)]
-pub struct AggregatedOrderbookEntry {
-    #[serde(flatten)]
-    pub entry: RpcOrderbookEntry,
-    #[serde(flatten)]
-    pub base_max_volume_aggr: AggregatedBaseVol,
-    #[serde(flatten)]
-    pub rel_max_volume_aggr: AggregatedRelVol,
+#[serde(tag = "method", rename = "orders_history_by_filter")]
+pub struct OrdersHistoryRequest {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub order_type: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub initial_action: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub base: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub rel: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub from_price: Option<MmNumber>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub to_price: Option<MmNumber>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub from_volume: Option<MmNumber>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub to_volume: Option<MmNumber>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub from_timestamp: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub to_timestamp: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub was_taker: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub status: Option<String>,
+    pub include_details: bool,
```

Comment:
Why remove `#[serde(default)]` from `include_details`, this breaks the logic of external tools that use the API if they don't provide this value. It's hard for reviewers when you remove such stuff while changing and moving the struct names as we have to look at each line of each struct to make sure nothing like this happened. Please be more careful with such things in the future and revise the code for any other instances like this.

Reply:
Thank you for pointing it out and sorry about it. 
[fixed](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1932/commits/f818cee4fce7a27c61fd9753f7ae7fc2c215f10c#diff-2d0339cc82af6222240226898743cc8cc53a28a22e5d6dd0c9e48ab8c6cb2933R443)

Reply:
Thank you for pointing it out and sorry about it. 
[fixed](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1932/commits/f818cee4fce7a27c61fd9753f7ae7fc2c215f10c#diff-2d0339cc82af6222240226898743cc8cc53a28a22e5d6dd0c9e48ab8c6cb2933R443)